### PR TITLE
Add Files Lifecycle occ command set

### PR DESCRIPTION
**Pull request by mmattel**
References: https://github.com/owncloud/docs/issues/3584 (occ command "lifecycle:" for files_lifecycle)

Add missing occ commands for `files_lifecycle`
Includes additional config:app commands to configure the policies.

Backport to 10.6 and 10.7

@jnweiger pls check if the occ command in section `Set Groups to be Excluded` is correct as I have not found an example to reference.

@cdamken fyi

### CODE REVIEW

**Comment by jnweiger** [comment]
Testing the behaviour of excluded_groups with files_lifecycle 1.3.1 on oc-1.7.0

image

```
occ config:app:set files_lifecycle excluded_groups --value="admin,user1"
occ lifecycle:archive -d
```
```
Start Archiving Process for all Users.
Running in dry run mode - no changes will be applied

    0 [>---------------------------]
Skipping user admin as they are except due to the policy.
Skipping user user1 as they are except due to the policy.
Moving file /user2/files/Photos/Portugal.jpg for user user2 to archive
Moving file /user2/files/Photos/Lake-Constance.jpg for user user2 to archive
Moving file /user2/files/Photos/Teotihuacan.jpg for user user2 to archive
Moving file /user2/files/Documents/Example.odt for user user2 to archive
Moving file /user2/files/ownCloud Manual.pdf for user user2 to archive
    7 [============================]

7 files would have been moved to archive.
```

File: modules/admin_manual/pages/configuration/server/occ_commands/app_commands/files_lifecycle.adoc [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+Restore all archived files for all users in the system back to their original locations. This command has no additional arguments or options.
+
+[source,console,subs="attributes+"]
+----
+{occ-command-example-prefix} lifecycle:restore-all
+----
+
+== Set Default Upload Time
+
+Set upload time for files which do not have one.
```
Comment (jnweiger) [inline modules/admin_manual/pages/configuration/server/occ_commands/app_commands/files_lifecycle.adoc:114]: Files without upload time are silently skipped by `occ lifecycle:archive`
E.g. files that were uploaded before the files_lifecycle app was configured (or when it is temporarily disabled) do not have an upload time.

**Review by jnweiger — CHANGES_REQUESTED** [review]
Suggest to hint at the importance of set-upload-time -- It may not be apparent, when this command is needed. Thus leading to possible frustration that files_lifecycle is not working properly.

File: modules/admin_manual/pages/configuration/server/occ_commands/app_commands/files_lifecycle.adoc (hunk 1 above)
Comment (mmattel) [reply modules/admin_manual/pages/configuration/server/occ_commands/app_commands/files_lifecycle.adoc:114]: fixed

**Comment by mmattel** [comment]
@jnweiger I have added the suggestion made + relocated some content for better understanding in the reading flow.  (eg. Archive Location in the FL document ect.) Please re-review and approve if you are fine with.

File: modules/admin_manual/pages/enterprise/file_management/files_lifecycle.adoc [hunk 2]
```diff
@@ -15,20 +15,39 @@ Depending on the desired level of enforcement, the extension provides two polici
...
+| `/<datadir>/$userid/archive/files/`
+|===
+
 === Setting Upload Times for Existing Files
 
 File Lifecycle Management uses the _upload time_ of files (server time at which it first appeared on the ownCloud server) to determine when to archive them. As ownCloud Server generally does not store this metadata, the File Lifecycle Management extension takes care of this when it is enabled.
 
-When File Lifecycle Management is set up on an existing ownCloud installation, you therefore have to set an _upload time_ for all files that existed before the extension has been enabled to allow the archiving policies to do their job.
-To set an upload time for all files that do not yet have one, you can use the _occ_ command `lifecycle:set-upload-time`. 
+When File Lifecycle Management is set up on an existing ownCloud installation, you therefore have to set an _upload time_ for all files that existed before the extension has been enabled or when it was temporarily disabled to allow the archiving policies to do their job. To set an upload time for all files that do not yet have one, you can use the _occ_ command `lifecycle:set-upload-time`.
```
Comment (EParzefall) [inline modules/admin_manual/pages/enterprise/file_management/files_lifecycle.adoc:46]:
```suggestion
When File Lifecycle Management is set up on an existing ownCloud installation, you therefore have to set an _upload time_ for all files that existed before the extension has been enabled. The same applies if it was temporarily disabled. Only then can the archiving policies work. To set an upload time for all files that do not yet have one, you can use the _occ_ command `lifecycle:set-upload-time`. 
```

File: modules/admin_manual/pages/enterprise/file_management/files_lifecycle.adoc [hunk 3]
```diff
@@ -15,20 +15,39 @@ Depending on the desired level of enforcement, the extension provides two polici
...
-To set an upload time for all files that do not yet have one, you can use the _occ_ command `lifecycle:set-upload-time`. 
+When File Lifecycle Management is set up on an existing ownCloud installation, you therefore have to set an _upload time_ for all files that existed before the extension has been enabled or when it was temporarily disabled to allow the archiving policies to do their job. To set an upload time for all files that do not yet have one, you can use the _occ_ command `lifecycle:set-upload-time`. 
 
 TIP: Files without an _upload time_ will not be considered for archiving.
 
-TIP: You only have to conduct this process once when setting up File Lifecycle Management on an installation with existing files. Files added after enabling File Lifecycle Management will be tracked automatically.
+TIP: You only have to conduct this process once when setting up File Lifecycle Management on an installation with existing files or when it was temporarily disabled . Files added after enabling File Lifecycle Management will be tracked automatically.
```
Comment (EParzefall) [inline modules/admin_manual/pages/enterprise/file_management/files_lifecycle.adoc:50]:
```suggestion
TIP: You only have to conduct this process once when setting up File Lifecycle Management on an installation with existing files or if it was temporarily disabled . Files added after enabling File Lifecycle Management will be tracked automatically.
```

**Review by EParzefall — APPROVED** [review]
Just a few language fixes.

**Pull request merged**